src/Processors/Formats/ISchemaReader.cpp

```diff
@@ -104,6 +115,26 @@ NamesAndTypesList IRowSchemaReader::readSchema()
             "Most likely setting input_format_max_rows_to_read_for_schema_inference is set to 0");
 
     DataTypes data_types = readRowAndGetDataTypes();
+    /// If column names weren't set, use default names 'c1', 'c2', ...
+    if (column_names.empty())
```

Comment:
In which case it is empty or non-empty?

Reply:
It's non-empty in two cases:
1) Format contains names and they were explicitely set using `setColumnNames` method. For example in formats with suffix `WithNames`: 
https://github.com/ClickHouse/ClickHouse/blob/110470809b288b44f0e7a9706c3280ae4ec8a5bc/src/Processors/Formats/RowInputFormatWithNamesAndTypes.cpp#L268
2) Column names were set using special settig `column_names_for_schema_inference`.